aiida/cmdline/params/types/path.py

```diff
@@ -31,3 +33,65 @@ def convert(self, value, param, ctx):
 
     def __repr__(self):
         return 'ABSOLUTEPATH'
+
+
+class ImportPath(click.Path):
+    """AiiDA extension of Click's Path-type to include URLs
+    An ImportPath can either be a `click.Path`-type or a URL.
+    """
+
+    # pylint: disable=protected-access
+
+    def __init__(self, timeout_seconds=None, **kwargs):
+        super(ImportPath, self).__init__(**kwargs)
+
+        # Set timeout_seconds
+        if timeout_seconds is None:
```

Comment:
What is the advantage of doing this instead of just putting it in the method signature? The default is immutable so that is fine?

Reply:
Reiterating a previous comment:

The default value is set in the constructor with a check if timeout_seconds is None set it to the default.
I think it is maybe better to keep a None value in the __init__ parameters, since if the default is set already here and someone should pass a None value the default value will still be used. Unless we specifically want it to raise if a None value is passed?

Reply:
According to our short talk, I'll simply define `timeout_seconds` in the `__init__` parameters.